### DISCUSSION TITLE
leafmap.array_to_image() gives "multiple values for argument 'transform'" error .

### DISCUSSION THREAD

**Discussion by Sjurk** [opengeos/leafmap#667] [Q&A] [answered]
I am trying to read and display an image from a NetCDF Dataset following https://leafmap.org/notebooks/89_image_array_viz/ :

### Code:
> ncds = Dataset("C:\lokal\pyproj\ODMviewer\output\distributed_elements_time_series_odm_hist.nc", 'r')  # netCDF4.Dataset
> temp = ncds.variables["sim_temperature"][:].filled(fill_value = 9.969209968386869e+36)   # temp is a 3D numpy array
> GT33 = (1000,0,-75000,0,-1000,8000000)                                             # (dx/dcol, dx/drow, EastEdge, dy/dcol, dy/drow, NorthEdge)
> tempimg = leafmap.array_to_image(array=temp[0,:], cellsize=1000, crs="EPSG:32633", transform=GT33)

Does anyone know what I can do to cure the error below?

Edit: By creating image by the leafmap.array_to_memory_file() function instead (parameters unchanged), I got rid of the error, but the geotransform seems to be ignored. 

The crs="EPSG32633" (UTM-33n) appears to be noticed; by changing the UTM zone I can move the image west to east with respect to the basemap. Specifying basemap={} hoping to avoid any reprojection did not du any other than removing the basemap.

### Output from original attempt, using leafmap.array_to_image:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_25472\3516671947.py in ?()
      1 ncds = Dataset("C:\lokal\pyproj\ODMviewer\output\distributed_elements_time_series_odm_hist.nc", 'r')
      2 temp = ncds.variables["sim_temperature"][:].filled(fill_value = 9.969209968386869e+36)
      3 GT33 = (1000,0,-75000,0,-1000,8000000)
      4 # tempimg = leafmap.array_to_image(array=temp[0,:], cellsize=0.01, crs="EPSG:4326")
----> 5 tempimg = leafmap.array_to_image(array=temp[0,:], cellsize=1000, crs="EPSG:32633", transform=GT33)
      6 print(type(temp), temp.shape, np.nanmean(temp), type(tempimg))

C:\Lokal\PyProj\ODMviewer\venv\Lib\site-packages\leafmap\common.py in ?(array, output, source, dtype, compress, transpose, cellsize, crs, driver, **kwargs)
  10408     import rasterio
  10409     import xarray as xr
  10410 
  10411     if output is None:
> 10412         return array_to_memory_file(
  10413             array, source, dtype, compress, transpose, cellsize, crs, driver, **kwargs
  10414         )
  10415 

TypeError: array_to_memory_file() got multiple values for argument 'transform'

### Output from updated attempt, using leafmap.array_to_memory_file(); Norway placed somewhere in Africa...

![image](https://github.com/opengeos/leafmap/assets/3890817/04cefe62-3e9d-4bc3-880b-6b8e2d6df05e)

**Comment by giswqs** [comment, marked as answer]
This issue has already been fixed in #660. I have not released a new version yet. You can update it using `leafmap.update_package()` and restart the kernel to take effect.

**Reply by Sjurk** [reply]
Excellent, thanks!  Both array_to_image() and array_to_memory_file worked as expected :-)
